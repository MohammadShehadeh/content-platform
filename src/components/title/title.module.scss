.title {
  font-weight: var(--font-weight-bold);
  line-height: 1.2;
  margin-bottom: var(--space-sm);
  text-transform: capitalize;
}

.sm {
  font-size: var(--font-size-sm);
}

.md {
  font-size: var(--font-size-md);
}

.lg {
  font-size: var(--font-size-lg);
}

.xl {
  font-size: calc(var(--font-size-lg) * 1.5);
}
