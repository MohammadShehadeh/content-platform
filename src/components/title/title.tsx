import { Slot } from '@radix-ui/react-slot';
import { cva, type VariantProps } from 'class-variance-authority';

import styles from './title.module.scss';

const titleVariants = cva(styles.title, {
  variants: {
    size: {
      sm: styles.sm,
      md: styles.md,
      lg: styles.lg,
      xl: styles.xl,
    },
  },
  defaultVariants: {
    size: 'lg',
  },
});

interface TitleProps
  extends React.HTMLAttributes<HTMLHeadingElement>,
    VariantProps<typeof titleVariants> {
  asChild?: boolean;
}

export const Title = ({
  children,
  className,
  asChild = false,
  size,
  ...props
}: TitleProps) => {
  const Comp = asChild ? Slot : 'h1';

  return (
    <Comp className={titleVariants({ size, className })} {...props}>
      {children}
    </Comp>
  );
};
