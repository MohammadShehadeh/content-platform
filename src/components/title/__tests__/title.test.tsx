import { render, screen } from '@testing-library/react';

import { Title } from '@/components/title';

describe('Title', () => {
  it('matches snapshot', () => {
    const { container } = render(<Title>Hello</Title>);

    expect(container).toMatchSnapshot();
  });

  it('should render with children', () => {
    render(<Title>Hello</Title>);
    expect(screen.getByText('Hello')).toBeInTheDocument();
  });

  it('should render with custom className', () => {
    render(<Title className="custom-class">Hello</Title>);
    expect(screen.getByText('Hello')).toHaveClass('custom-class');
  });

  it('should render with different size variants', () => {
    render(<Title size="sm">Small</Title>);
    render(<Title size="md">Medium</Title>);
    render(<Title size="lg">Large</Title>);

    expect(screen.getByText('Small')).toHaveClass('sm');
    expect(screen.getByText('Medium')).toHaveClass('md');
    expect(screen.getByText('Large')).toHaveClass('lg');
  });
});
