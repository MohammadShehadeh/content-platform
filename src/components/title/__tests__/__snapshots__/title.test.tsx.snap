// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title matches snapshot 1`] = `
<div>
  <h1
    class="title lg"
  >
    Hello
  </h1>
</div>
`;
